### DISCUSSION TITLE
Get problem when updating OpenSSL from 1.11.1u to 3.0.11.

### DISCUSSION THREAD

**Discussion by SAO-kirito-asuna** [openssl/openssl#22451] [Q&A]
I'm working on updating openssl from 1.11.1u to 3.0.11.I downloaded the code for the openssl 3.0.11 branch.And use the following 
`perl Configure VC-WIN32 no-asm --prefix=%openssl_output_x86%
nmake clean
nmake
nmake install_sw`
I got these components:libcrypto.lib,libssl.lib,libcrypto-3.dll,libssl-3.dll.
I also updated Poco with Openssl3.0.11.
Then I ran my program and evenything is okay.So I packed the program the send it to another computer.But it can't run properly.It throw the exception when calling the Poco::Net::initializeSSL():
Microsoft C++ exception: Poco::Crypto::CryptoException at memory location 0x0193F930.
Why is this happening?

**Comment by SAO-kirito-asuna** [comment]
The commands are the following:
perl Configure VC-WIN32 no-asm --prefix=%openssl_output_x86%
nmake clean
nmake
nmake install_sw

**Reply by kulkarniamit** [reply]
What version of poco library are you using? May be it's related to https://github.com/pocoproject/poco/issues/4077 ?

**Reply by SAO-kirito-asuna** [reply]
The version of poco is 1.12.4.

**Reply by SAO-kirito-asuna** [reply]
What is legacy provider?What happens if the legacy provider fails to be loaded?

**Reply by paulidale** [reply]
https://www.openssl.org/docs/man3.0/man7/OSSL_PROVIDER-legacy.html

If it fails to load those algorithms it provides are not available.

**Reply by SAO-kirito-asuna** [reply]
Does OpenSSL3.0.11 support TLS1.0?
m_ctx = SSL_CTX_new(TLSv1_method());
SSL_CTX_set_verify(m_ctx, SSL_VERIFY_NONE | SSL_VERIFY_FAIL_IF_NO_PEER_CERT, nullptr);
m_ssl = SSL_new(m_ctx);
long options = SSL_OP_NO_SSLv2 | SSL_OP_NO_SSLv3 | SSL_OP_SINGLE_DH_USE | SSL_OP_DONT_INSERT_EMPTY_FRAGMENTS;
SSL_set_options(m_ssl, options);
int nerr = SSL_connect(m_ssl);
I try to establish an SSL connection.But the nerr  is -1;The error message is:
error:0A00014D:SSL routines::legacy sigalg disallowed or unsupported
What shoud I do?

**Reply by t8m** [reply]
You need to set the security level to 0 for TLS1.0. Set the cipher string to `DEFAULT:@SECLEVEL=0`

**Reply by SAO-kirito-asuna** [reply]
How to set the cipher string to DEFAULT:@SECLEVEL=0 in Visual studio 2015?

**Reply by slontis** [reply]
````
SSL_set_cipher_list(m_ssl, "DEFAULT:@SECLEVEL=0");
````
OR use 
````
SSL_CTX_set_cipher_list(m_ctx, "DEFAULT:@SECLEVEL=0");
````

**Comment by SAO-kirito-asuna** [comment]
Failed to load legacy provider using OSSL_PROVIDER_load(NULL, "legacy") . Is there something missing?

**Comment by slontis** [comment]
The legacy.dll file is a provider that supplies all old deprecated algorithms. You would need to copy this file..
(You may also have to set the path via OPENSSL_MODULES)
